x-pack/plugins/enterprise_search/server/lib/indices/pipelines/ml_inference/update_ml_inference_mappings.ts

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ElasticsearchClient } from '@kbn/core-elasticsearch-server';
+
+import { ErrorCode } from '../../../../../common/types/error_codes';
+
+import { isNotNullish } from '../../../../../common/utils/is_not_nullish';
+import { isIllegalArgumentException } from '../../../../utils/identify_exceptions';
+
+/**
+ * Creates Elasticsearch field mappings so that the outputs of ML Inference pipelines are not indexed as `float` fields
+ * @param indexName - the index whose mapping will be updated
+ * @param fieldMappings - the object representing the source field (text) names and target fields (ML output) names
+ * @param esClient
+ */
+export const updateMlInferenceMappings = async (
+  indexName: string,
+  fieldMappings: Record<string, string | undefined>,
+  esClient: ElasticsearchClient
+) => {
+  // Today, we only update mappings for ELSER-analyzed fields.
+  const mapping = formELSERMappingProperties(
+    Object.keys(fieldMappings),
+    Object.values(fieldMappings).filter(isNotNullish)
+  );
+  try {
+    return await esClient.indices.putMapping({
+      index: indexName,
+      properties: mapping,
+    });
+  } catch (e) {
+    if (isIllegalArgumentException(e)) {
+      throw new Error(ErrorCode.MAPPING_UPDATE_FAILED);
+    } else {
+      throw e;
+    }
+  }
+};
+
+const formELSERMappingProperties = (sourceFields: string[], targetFields: string[]) => {
```

Comment:
Nit: I think acronyms in function names follow camelCase notation (see [Microsoft's style guide](https://learn.microsoft.com/en-us/previous-versions/dotnet/netframework-1.1/141e06ef(v=vs.71)?redirectedfrom=MSDN) -> `HtmlButton`), but I'm not sure what our team's guidelines are for this.
```suggestion
const formElserMappingProperties = (sourceFields: string[], targetFields: string[]) => {
```